packages/lib/server/defaultResponder.ts

```diff
@@ -13,7 +13,9 @@ export function defaultResponder<T>(f: Handle<T>) {
       performance.mark("Start");
       const result = await f(req, res);
       ok = true;
-      if (result) res.json(result);
+      if (result && !res.writableEnded) {
```

Comment:
There could be cases where header is already set and result is also set. e.g.` res.redirect` unlike `res.json` returns a non nullish value and thus we would endup setting `json` in that case as well which would cause 'header already set error'

Reply:
Nice find. Unit testable?